MM2RandoLib/Utilities/MiscHacks.cs

```diff
@@ -558,8 +682,244 @@ public static void ReduceUnderwaterLag(Patch p)
         /// <param name="p"></param>
         public static void DisableDelayScroll(Patch p)
         {
-            p.Add((Int32)ESubroutineAddress.ChangeBankBNE, (Byte)EInstruction.NOP, "Disable the delayed audio processing branch");
-            p.Add((Int32)ESubroutineAddress.ChangeBankBNE + 1, (Byte)EInstruction.NOP, "The branch instruction is 2 bytes");
+            p.Add((Int32)ESubroutineAddress.ChangeBankBNE, Opcode6502.NOP, "Disable the delayed audio processing branch");
+            p.Add((Int32)ESubroutineAddress.ChangeBankBNE + 1, Opcode6502.NOP, "The branch instruction is 2 bytes");
+        }
+
+
+        /// <summary>
+        /// This method will add a new subroutine to the ROM which triggers
+        /// enemy and item spawns in Wily 5.
+        ///
+        /// At 0x0E:816A (bank E, 0x3817A in the ROM), there is a CMP
+        /// instruction to value 0x0C (CMP #0x0C). This is checking if the
+        /// current stage is Wily 5 (0x0C).  If the current stage is Wily 5,
+        /// it jumps to a special game loop at 0x0E:8223 (bank E, 0x38233
+        /// in the ROM), which has a special setup routine for the teleport
+        /// room, otherwise, it jumps to 0x0E:8171, which is the normal game loop.
+        /// The special Wily 5 game loop does not include the jump instruction
+        /// to the subroutine that spawns items or enemies.
+        ///
+        /// This method will insert a new special Wily 5 game loop at the end
+        /// of the data in bank E, which does include the item spawn routine,
+        /// so that weapon energy pickups will spawn.
+        ///
+        /// The jump to the special Wily 5 routine will be changed from
+        /// 0x0E:8223 to 0x0E:BD24 (bank E, 0x3BD37 in the ROM).
+        ///
+        /// The new special Wily 5 routine will have the jump to the item spawn
+        /// routine at 0x0E:D658.
+        /// </summary>
+        public static void AddWily5SubroutineWithItemSpawns(Patch p)
```

Comment:
Similar to the other randomization changes, I'm not sure these make sense in the context of the current PR. They feel like they should be in a separate PR.